src/Module/OAuth/Authorize.php

```diff
@@ -52,7 +50,9 @@ public static function rawContent(array $parameters = [])
 		$uid = local_user();
 		if (empty($uid)) {
 			Logger::info('Redirect to login');
-			DI::app()->redirect('login?return_path=/oauth/authorize');
+			$request = $_REQUEST;
+			unset($request['pagename']);
+			DI::app()->redirect('login?return_path=/oauth/authorize' . urlencode('?' . http_build_query($request)));
```

Comment:
This looks incorrect, the `?` is already present in the redirect URL. Additionally, `http_build_query()` applies `urlencode()`, so this would lead to double-encoding.
```suggestion
			$request['return_path'] = '/oauth/authorize';
			DI::app()->redirect('login?' . http_build_query($request));
```

Reply:
Oh, I see, so I'd prefer it to be something like this so that it's clearer the `$request` parameters are used in the URL we provided as a parameter.
```suggestion
			DI::app()->redirect('login?' . http_build_query(['return_path' => '/oauth/authorize?' . http_build_query($request)]));
```

Reply:
Or simpler:
```suggestion
			DI::app()->redirect('login?return_path=' . urlencode('/oauth/authorize?' . http_build_query($request)));
```